packages/amplify-graphql-api-construct/src/amplify-graphql-definition.ts

```diff
@@ -88,11 +89,38 @@ export class AmplifyGraphqlDefinition {
     if (definitions.length === 1) {
       return definitions[0];
     }
+
+    // A strategy will be present multiple times in a given definition: once per model. We'll create a unique list per definition to ensure
+    // no reuse across definitions.
+    let combinedStrategyNames: string[] = [];
+    for (const definition of definitions) {
+      const definitionStrategyNames = new Set<string>();
+      for (const strategy of Object.values(definition.dataSourceStrategies)) {
+        if (!isSqlStrategy(strategy)) {
+          continue;
+        }
+        const strategyName = strategy.name;
+        if (combinedStrategyNames.includes(strategyName)) {
+          throw new Error(
+            `A SQL-based ModelDataSourceStrategy name ('${strategyName}') cannot be shared between definitions. To specify a SQL-based API with schemas across multiple files, use 'fromFilesAndStrategy'`,
```

Comment:
nit: 
```suggestion
            `A SQL-based ModelDataSourceStrategy named ('${strategyName}') cannot be shared between definitions. To specify a SQL-based API with schemas across multiple files, use 'fromFilesAndStrategy'`,
```

Reply:
Hmm. The sentence is, "A name cannot be shared", I was trying to give context on which name was duped. I'll change to `The SQL-based ModelDataSourceStrategy ('${strategyName}') was found in multiple definitions, but a strategy name cannot be shared between definitions. To specify...`

Reply:
Fixed in f2774ee62